main.go

```diff
@@ -139,6 +141,8 @@ func main() {
 		"The duration of time that artifacts will be kept in storage before being garbage collected.")
 	flag.IntVar(&artifactRetentionRecords, "artifact-retention-records", 2,
 		"The maximum number of artifacts to be kept in storage after a garbage collection.")
+	flag.Int64Var(&artifactMaxSize, "artifact-max-size", artifactMaxSizeDefault,
+		"The max allowed size in bytes of an artifact contents produced from sources. The limit can be disabled by setting the value to -1.")
```

Comment:
By reading the code I understand that any value <= 0 will disable the limit. But on the flag here we say `-1` explicitly and in the tests we use `0`. Maybe it'd be better to use a fixed value everywhere, either -1 or 0, or make it clear in the flag that any value <= 0 disables the limit.